### DISCUSSION TITLE
How to pull a list of accounts and the number of hosts/agents for each one...

### DISCUSSION THREAD

**Discussion by DougBalensiefen** [CrowdStrike/falconpy#1184] [Q&A]
I want to build a simple tool that pulls a list of accounts on 1 of 2 base_urls and their subscription info (max number of hosts, correct number of hosts).  I have looked through the docs but am not finding a way to do this.  I have tried "query_accounts" and "get_accounts" but they do not return what I am looking for.  Any direction here would he helpful.  Thanks, in advance, for any help.

**Comment by David-M-Berry** [comment]
Hi @DougBalensiefen  - when you say a list of accounts and the number of hosts/agents, I take that to mean you are meaning at the CID/tenant level, and then their subscription allotments? 

If that's the case there is not yet a billing/subscription API. You would need to already be aware of your subscription allotment and use the host/device APIs to check how many were there. Also, your API client is going to need to be different/separate for each CID/tenant.

**Comment by DougBalensiefen** [comment]
We are multi-tenant with several clients.  I would like to be able to loop through accounts and determine the number of hosts/devices each client is using.  Based on your answer, it seems I just need to figure out how to get to each client, then I can query the hosts for them and determine the count.

**Reply by David-M-Berry** [reply]
Correct. As each API client is CID/tenant specific, if you can keep their output concise on your side, it should be "reconcilable". :)

**Comment by DougBalensiefen** [comment]
Thanks for the assistance.  I made some progress yesterday but still have
an issue I haven't figured out yet.  If I use Discover to get the host
info, I get a total number of devices that is slightly (in most cases)
higher than the "Sensor Usage" number that shows up in the console.  I am
wondering if anyone knows which fields are used to calculated the sensor
usage number...hoping I can build an FQL query to filter only for those.
Any ideas would be appreciated.
